src/mono/mono/mini/interp/transform.c

```diff
@@ -8366,49 +8375,117 @@ interp_mark_reachable_bblocks (TransformData *td)
 	}
 }
 
+/**
+ * Returns TRUE if in_bb defines a variable that is used in
+ * the block containing @cond_ins, FALSE otherwise.
+ */
+
 static gboolean
-interp_prev_ins_defines_var (InterpInst *ins, int var1, int var2)
+interp_prev_block_defines_var (InterpInst *in_bb_ins, InterpInst *cond_ins)
 {
+	InterpInst *prev_ins = interp_prev_ins(cond_ins);
+	int var1 = -1, var2 = -1;
```

Comment:
I feel like having the lookup here is not really the best. It doesn't correctly (strictly speaking) initializes the lookup vars. For example in some of the patterns, we know the value of one of the sregs from the existing bblock (`ldc` + `compare`/`condbr`), in which case we should only look for the unkown var. Also we can set these lookup vars directly in `interp_inline_into_callers`, as we check the patterns, without having these separate checks.

Reply:
Good idea, updated.